### ISSUE TITLE
RestrictEvents reads boot-args revcpu and revcpuname but chooses CPU name from elsewhere

### ISSUE THREAD

**Issue by whatnameisit** [acidanthera/bugtracker#1727]
As the title says, RestrictEvents reads dedicated boot-args, but the CPU name is not patched. Attaching Lilu log and my config below. 
[Lilu_1.5.4_17.7.txt](https://github.com/acidanthera/bugtracker/files/6853663/Lilu_1.5.4_17.7.txt)
[config.plist.zip](https://github.com/acidanthera/bugtracker/files/6853695/config.plist.zip)
Also attaching System Information.
![스크린샷 2021-07-21 오후 4 47 35](https://user-images.githubusercontent.com/47713349/126451516-9b4d0ac8-8877-4a67-b62c-3bf77f82bc4e.png)

**Comment by vit9696** [comment]
dmidecode output?

**Comment by whatnameisit** [comment]
Do I need to use an argument with dmidecode? If not, below is the output on terminal by opening the binary.
```
Last login: Thu Jul 22 10:05:46 on ttys000
/Users/toshiba/Downloads/dmidecode ; exit;
Toshibas-MacBook-Pro:~ toshiba$ /Users/toshiba/Downloads/dmidecode ; exit;
# dmidecode 3.3b
Scanning I/O Registry for entry point.
SMBIOS 3.2 present.
20 structures occupying 1170 bytes.
Table at 0xAF392000.

Handle 0x0001, DMI type 0, 26 bytes
0000: 00 1A 01 00 01 02 00 00 03 3F 80 9A 09 48 00 00
0010: 01 00 87 07 01 1E 01 28 00 00

BIOS Information
	Vendor: Acidanthera
	Version: 421.0.0.0.0
	Release Date: 12/17/2020
	ROM Size: 4 MB
	Characteristics:
		PCI is supported
		PNP is supported
		BIOS is upgradeable
		BIOS shadowing is allowed
		Boot from CD is supported
		Selectable boot is supported
		EDD is supported
		8042 keyboard services are supported (int 9h)
		CGA/mono video services are supported (int 10h)
		ACPI is supported
		USB legacy is supported
		AGP is supported
		Smart battery is supported
		BIOS boot specification is supported
		Function key-initiated network boot is supported
		Targeted content distribution is supported
	BIOS Revision: 1.30
	Firmware Revision: 1.40

Handle 0x0002, DMI type 1, 27 bytes
0000: 01 1B 02 00 01 02 03 04 23 EB 4D 8C 1D 07 26 47
0010: B0 C7 1E 21 99 17 3F EF 06 05 06

System Information
	Manufacturer: Acidanthera
	Product Name: MacBookPro9,1
	Version: 1.0
	Serial Number: C02KK0SCF1G4
	UUID: {23 eb 4d 8c 1d 07 26 47 b0 c7 1e 21 99 17 3f ef} (RAW)
	UUID: 8c4deb23-071d-4726-b0c7-1e2199173fef (SMBIOS 2.6+, preferred)
	UUID: 23eb4d8c-1d07-2647-b0c7-1e2199173fef (RFC 4122, Apple)
	Wake-up Type: Power Switch
	SKU Number: System SKU#
	Family: MacBook Pro

Handle 0x0003, DMI type 2, 15 bytes
0000: 02 0F 03 00 01 02 03 04 05 00 06 04 00 0A 00

Base Board Information
	Manufacturer: Acidanthera
	Product Name: Mac-4B7AC7E43945597E
	Version: MacBookPro9,1
	Serial Number: C02315902J9F327UE
	Asset Tag: Base Board Asset Tag#
	Features: None
	Location In Chassis: Part Component
	Chassis Handle: 0x0004
	Type: Motherboard
	Contained Object Handles: 0

Handle 0x0004, DMI type 3, 22 bytes
0000: 03 16 04 00 01 0A 02 03 00 03 03 03 03 00 00 00
0010: 00 00 01 00 00 00

Chassis Information
	Manufacturer: Acidanthera
	Type: Notebook
	Lock: Not Present
	Version: Mac-4B7AC7E43945597E
	Serial Number: C02KK0SCF1G4
	Asset Tag: Not Specified
	Boot-up State: Safe
	Power Supply State: Safe
	Thermal State: Safe
	Security Status: None
	OEM Information: 0x00000000
	Height: Unspecified
	Number Of Power Cords: 1
	Contained Elements: 0
	SKU Number: Not Specified

Handle 0x0080, DMI type 7, 27 bytes
0000: 07 1B 80 00 01 80 00 20 00 20 00 20 00 20 00 00
0010: 05 03 07 20 00 00 00 20 00 00 00

Cache Information
	Socket Designation: L1 Cache
	Configuration: Enabled, Not Socketed, Level 1
	Operational Mode: Write Through
	Location: Internal
	Installed Size: 32 kB
	Maximum Size: 32 kB
	Supported SRAM Types:
		Synchronous
	Installed SRAM Type: Synchronous
	Speed: Unknown
	Error Correction Type: Single-bit ECC
	System Type: Instruction
	Associativity: 8-way Set-associative

Handle 0x0081, DMI type 7, 27 bytes
0000: 07 1B 81 00 01 81 00 00 01 00 01 20 00 20 00 00
0010: 05 05 07 00 01 00 00 00 01 00 00

Cache Information
	Socket Designation: L2 Cache
	Configuration: Enabled, Not Socketed, Level 2
	Operational Mode: Write Through
	Location: Internal
	Installed Size: 256 kB
	Maximum Size: 256 kB
	Supported SRAM Types:
		Synchronous
	Installed SRAM Type: Synchronous
	Speed: Unknown
	Error Correction Type: Single-bit ECC
	System Type: Unified
	Associativity: 8-way Set-associative

Handle 0x0082, DMI type 7, 27 bytes
0000: 07 1B 82 00 01 82 00 20 80 20 80 20 00 20 00 00
0010: 05 05 01 20 00 00 80 20 00 00 80

Cache Information
	Socket Designation: L3 Cache
	Configuration: Enabled, Not Socketed, Level 3
	Operational Mode: Write Through
	Location: Internal
	Installed Size: 2 MB
	Maximum Size: 2 MB
	Supported SRAM Types:
		Synchronous
	Installed SRAM Type: Synchronous
	Speed: Unknown
	Error Correction Type: Single-bit ECC
	System Type: Unified
	Associativity: Other

Handle 0x0083, DMI type 4, 48 bytes
0000: 04 30 83 00 01 03 0B 02 A7 06 02 00 FF FB EB BF
0010: 03 8D 64 00 D0 07 D0 07 41 04 80 00 81 00 82 00
0020: 00 04 00 02 02 02 04 00 0B 00 02 00 02 00 02 00

Processor Information
	Socket Designation: CPU
	Type: Central Processor
	Family: Pentium
	Manufacturer: Intel(R) Corporation
	ID: A7 06 02 00 FF FB EB BF
	Signature: Type 0, Family 6, Model 42, Stepping 7
	Flags:
		FPU (Floating-point unit on-chip)
		VME (Virtual mode extension)
		DE (Debugging extension)
		PSE (Page size extension)
		TSC (Time stamp counter)
		MSR (Model specific registers)
		PAE (Physical address extension)
		MCE (Machine check exception)
		CX8 (CMPXCHG8 instruction supported)
		APIC (On-chip APIC hardware supported)
		SEP (Fast system call)
		MTRR (Memory type range registers)
		PGE (Page global enable)
		MCA (Machine check architecture)
		CMOV (Conditional move instruction supported)
		PAT (Page attribute table)
		PSE-36 (36-bit page size extension)
		CLFSH (CLFLUSH instruction supported)
		DS (Debug store)
		ACPI (ACPI supported)
		MMX (MMX technology supported)
		FXSR (FXSAVE and FXSTOR instructions supported)
		SSE (Streaming SIMD extensions)
		SSE2 (Streaming SIMD extensions 2)
		SS (Self-snoop)
		HTT (Multi-threading)
		TM (Thermal monitor supported)
		PBE (Pending break enabled)
	Version: Intel(R) Pentium(R) CPU B940 @ 2.00GHz
	Voltage: 1.3 V
	External Clock: 100 MHz
	Max Speed: 2000 MHz
	Current Speed: 2000 MHz
	Status: Populated, Enabled
	Upgrade: ZIF Socket
	L1 Cache Handle: 0x0080
	L2 Cache Handle: 0x0081
	L3 Cache Handle: 0x0082
	Serial Number: Not Specified
	Asset Tag: FFFF
	Part Number: Not Specified
	Core Count: 2
	Core Enabled: 2
	Thread Count: 2
	Characteristics:
		64-bit capable

Handle 0x0084, DMI type 16, 23 bytes
0000: 10 17 84 00 03 03 03 00 00 00 02 FF FF 02 00 00
0010: 00 00 00 00 00 00 00

Physical Memory Array
	Location: System Board Or Motherboard
	Use: System Memory
	Error Correction Type: None
	Maximum Capacity: 32 GB
	Error Information Handle: No Error
	Number Of Devices: 2

Handle 0x0085, DMI type 19, 31 bytes
0000: 13 1F 85 00 00 00 00 00 FF FF 7F 00 84 00 02 00
0010: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00

Memory Array Mapped Address
	Starting Address: 0x00000000000
	Ending Address: 0x001FFFFFFFF
	Range Size: 8 GB
	Physical Array Handle: 0x0084
	Partition Width: 2

Handle 0x0086, DMI type 17, 92 bytes
0000: 11 5C 86 00 84 00 FF FF 40 00 40 00 00 10 0D 00
0010: 01 02 18 80 00 35 05 03 04 05 06 00 00 00 00 00
0020: 35 05 00 00 00 00 00 00 00 00 00 00 00 00 00 00
0030: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
0040: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
0050: 00 00 00 00 00 00 00 00 00 00 00 00

Memory Device
	Array Handle: 0x0084
	Error Information Handle: No Error
	Total Width: 64 bits
	Data Width: 64 bits
	Size: 4 GB
	Form Factor: SODIMM
	Set: None
	Locator: DIMM0
	Bank Locator: BANK 0
	Type: DDR3
	Type Detail: Synchronous
	Speed: 1333 MT/s
	Manufacturer: Samsung
	Serial Number: 6101F0CC
	Asset Tag: 0123456789
	Part Number: M471B5273CH0-CH9
	Rank: Unknown
	Configured Memory Speed: 1333 MT/s
	Minimum Voltage: Unknown
	Maximum Voltage: Unknown
	Configured Voltage: Unknown
	Memory Technology: <OUT OF SPEC>
	Memory Operating Mode Capability: None
	Firmware Version: Not Specified
	Module Manufacturer ID: Unknown
	Module Product ID: Unknown
	Memory Subsystem Controller Manufacturer ID: Unknown
	Memory Subsystem Controller Product ID: Unknown
	Non-Volatile Size: None
	Volatile Size: None
	Cache Size: None
	Logical Size: None

Handle 0x0087, DMI type 20, 35 bytes
0000: 14 23 87 00 00 00 00 00 FF FF 3F 00 86 00 85 00
0010: FF 01 01 00 00 00 00 00 00 00 00 00 00 00 00 00
0020: 00 00 00

Memory Device Mapped Address
	Starting Address: 0x00000000000
	Ending Address: 0x000FFFFFFFF
	Range Size: 4 GB
	Physical Device Handle: 0x0086
	Memory Array Mapped Address Handle: 0x0085
	Partition Row Position: Unknown
	Interleave Position: 1
	Interleaved Data Depth: 1

Handle 0x0088, DMI type 17, 92 bytes
0000: 11 5C 88 00 84 00 FF FF 40 00 40 00 00 10 0D 00
0010: 01 02 18 80 00 35 05 03 04 05 06 00 00 00 00 00
0020: 35 05 00 00 00 00 00 00 00 00 00 00 00 00 00 00
0030: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
0040: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
0050: 00 00 00 00 00 00 00 00 00 00 00 00

Memory Device
	Array Handle: 0x0084
	Error Information Handle: No Error
	Total Width: 64 bits
	Data Width: 64 bits
	Size: 4 GB
	Form Factor: SODIMM
	Set: None
	Locator: DIMM1
	Bank Locator: BANK 2
	Type: DDR3
	Type Detail: Synchronous
	Speed: 1333 MT/s
	Manufacturer: Samsung
	Serial Number: 6736581C
	Asset Tag: 0123456789
	Part Number: M471B5273CH0-CH9
	Rank: Unknown
	Configured Memory Speed: 1333 MT/s
	Minimum Voltage: Unknown
	Maximum Voltage: Unknown
	Configured Voltage: Unknown
	Memory Technology: <OUT OF SPEC>
	Memory Operating Mode Capability: None
	Firmware Version: Not Specified
	Module Manufacturer ID: Unknown
	Module Product ID: Unknown
	Memory Subsystem Controller Manufacturer ID: Unknown
	Memory Subsystem Controller Product ID: Unknown
	Non-Volatile Size: None
	Volatile Size: None
	Cache Size: None
	Logical Size: None

Handle 0x0089, DMI type 20, 35 bytes
0000: 14 23 89 00 00 00 40 00 FF FF 7F 00 88 00 85 00
0010: FF 02 01 00 00 00 00 00 00 00 00 00 00 00 00 00
0020: 00 00 00

Memory Device Mapped Address
	Starting Address: 0x00100000000
	Ending Address: 0x001FFFFFFFF
	Range Size: 4 GB
	Physical Device Handle: 0x0088
	Memory Array Mapped Address Handle: 0x0085
	Partition Row Position: Unknown
	Interleave Position: 2
	Interleaved Data Depth: 1

Handle 0x0019, DMI type 32, 11 bytes
0000: 20 0B 19 00 00 00 00 00 00 00 00

System Boot Information
	Status: No errors detected

Handle 0x0029, DMI type 131, 8 bytes
0000: 83 08 29 00 01 06 00 00

Apple specific Processor Type
	CPU: 0x0601

Handle 0x0027, DMI type 128, 96 bytes
0000: 80 60 27 00 00 00 00 00 37 E1 0D C0 3F FF 1F FF
0010: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
0020: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
0030: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
0040: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
0050: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00

Apple specific FirmwareVolume
	FirmwareFeatures: 0xC00DE137
	FirmwareFeaturesMask: 0xFF1FFF3F

Handle 0x002B, DMI type 133, 12 bytes
0000: 85 0C 2B 00 00 00 00 00 00 00 00 00

Apple specific PlatformFeature
	PlatformFeature: 0x00000000

Handle 0x002C, DMI type 134, 20 bytes
0000: 86 14 2C 00 32 2E 31 46 30 31 37 35 00 00 00 00
0010: 00 00 00 00

Apple specific SmcVersion
	SmcVersion: 2.1F0175

Handle 0x0026, DMI type 127, 4 bytes
0000: 7F 04 26 00

End Of Table

logout
Saving session...
...copying shared history...
...saving history...truncating history files...
...completed.
Deleting expired sessions...6 completed.

[프로세스 완료됨]

```

**Comment by Andrey1970AppleLife** [comment]
Wrong value.
![Снимок экрана 2021-07-22 в 04 55 11](https://user-images.githubusercontent.com/17758753/126580642-7769cc16-db67-4838-8066-831e0b8bb636.png)
Value boot-args **revcpuname** shan't begin with digit. Specify values since a letter.
If you need to begin value with digit, use NVRAM variables.

**Comment by whatnameisit** [comment]
That's something I didn't know.
Still revcpuname isn't correctly applied. This time I put it under NVRAM variable 4D1FDA02-38C7-4A6A-9CC6-4BCCA8B30102:revcpuname.
[Lilu_1.5.4_17.7.txt](https://github.com/acidanthera/bugtracker/files/6859823/Lilu_1.5.4_17.7.txt)

**Comment by vit9696** [comment]
Can you show what you see in About Mac? Ideally in English.

Try replacing these lines:
https://github.com/acidanthera/RestrictEvents/blob/master/RestrictEvents/RestrictEvents.cpp#L234-L235

with:

```C++
				cpuFindPatch = "\0" "Intel Core i5";
				cpuFindSize = sizeof("\0" "Intel Core i5");
```

I get a feeling Dual-Core prefix is unused for you for whatever reason.

**Comment by whatnameisit** [comment]
Replacing those lines made About this Mac show the name I chose.
If you still need them, here are the screenshots.
Before
![스크린샷 2021-07-22 3 46 36 PM](https://user-images.githubusercontent.com/47713349/126600597-8afe4664-2dd5-442e-9685-1a4449a8c834.png)
After
![스크린샷 2021-07-22 3 55 50 PM](https://user-images.githubusercontent.com/47713349/126600616-ecf6a018-5519-468f-8f45-5a788eb95d3f.png)

**Comment by vit9696** [comment]
Hmmm, this is 10.13, and it might be the reason it failed. Can you check 10.14, 10.15, and 11?

**Comment by whatnameisit** [comment]
Will try next week. I've left the laptop at my workplace.
By "it" do you mean the Release kext might patch CPU name, the same kext will have no spaces in front of revcpu default CPU name, or the modified kext?

**Comment by whatnameisit** [comment]
I tested 10.13 and 11 on a different laptop and it does fail on 10.13. Different OS versions use different CPU names.
[Archive.zip](https://github.com/acidanthera/bugtracker/files/6868459/Archive.zip)

Is this enough or do you want me to test 10.14 and 10.15 as well? Out of laziness I Googled "Mojave About this Mac" and Catalina and it seems the naming changed since Catalina.

**Comment by vit9696** [comment]
Yes, basically need to know when this started to happen to make a change only apply on older macOS versions.

**Comment by whatnameisit** [comment]
What about this?

> The CPU name starts with five spaces when not specifying revcpuname. Can this be fixed, or is it my laptop's fault?
> ![Screen Shot 2021-07-22 at 4 05 17 PM](https://user-images.githubusercontent.com/47713349/126601645-2c41d76e-f17c-4c44-a238-5ab9f016dca0.png)
> [Lilu_1.5.4_17.7.txt](https://github.com/acidanthera/bugtracker/files/6860507/Lilu_1.5.4_17.7.txt)

And by the way, I just realized that the attachment in my previous comment were obtained with ProcessorType=0, not 1537. Do I need to obtain the logs again with 1537? The laptop has an i5 processor. I did set it to 1537 in the other laptop in the opening issue. Now I'm having some difficulty understanding what this means: `Note: Apple CPU identifier must be 0x0F01 for 8 core CPUs or higher and 0x0601 for 1, 2, 4, or 6 cores. This is the default in OpenCore for non-natively supported CPUs.`

**Comment by vit9696** [comment]
Well, your cpu already does happen to default to it in OC.

**Comment by whatnameisit** [comment]
Oh, I get it now.

Here's the rest of the logs and About this Mac screenshots.
[Archive.zip](https://github.com/acidanthera/bugtracker/files/6872128/Archive.zip)

Anything for multiple spacing prefix in CPU name in High Sierra?

**Comment by whatnameisit** [comment]
Pentium 3558U seems it would not have leading spaces.
[Lilu_1.5.4_17.7.txt](https://github.com/acidanthera/bugtracker/files/6873793/Lilu_1.5.4_17.7.txt)

**Comment by vit9696** [comment]
Thanks! Please confirm that master version works for you.

**Comment by vit9696** [comment]
> Pentium 3558U seems it would not have leading spaces.

Did not get it.

**Comment by whatnameisit** [comment]
> > Pentium 3558U seems it would not have leading spaces.
> 
> Did not get it.

I meant this.

> ![Screen Shot 2021-07-22 at 4 05 17 PM](https://user-images.githubusercontent.com/47713349/126601645-2c41d76e-f17c-4c44-a238-5ab9f016dca0.png)
> [Lilu_1.5.4_17.7.txt](https://github.com/acidanthera/bugtracker/files/6860507/Lilu_1.5.4_17.7.txt)

**Comment by whatnameisit** [comment]
I'm yet to test it on the laptop that I raised this issue with. I'm doing some repartitioning on the laptop that I debugged 10.13-11 on, so I can't use that either. On a different machine with Pentium 3558U the master version failed on 10.13.
Pentium 3558U 10.13
[Pentium 3558U.zip](https://github.com/acidanthera/bugtracker/files/6873920/Pentium.3558U.zip)
Pentium 3558U 11
yet

**Comment by vit9696** [comment]
> I meant this.

Should be fixed in master.

> Pentium 3558U 10.13 Doesn't work.

Are you sure this is the latest version (i.e. compiled from master)?

**Comment by whatnameisit** [comment]
Yes, I downloaded the artifacts.

**Comment by whatnameisit** [comment]
The CPU name patch is processed on the B940 laptop. But leading spaces are still there. It seems they are removed in the log compared to previous logs, though.
[Leading spaces with f4464e7.txt](https://github.com/acidanthera/bugtracker/files/6875119/Leading.spaces.with.f4464e7.txt)
![Leading spaces with  f4464e7 ](https://user-images.githubusercontent.com/47713349/126924312-81993f8a-3ee2-4e6b-be66-e792a65c24e3.png)

**Comment by vit9696** [comment]
> The CPU name patch is processed on the B940 laptop. But leading spaces are still there. It seems they are removed in the log compared to previous logs, though.

Indeed. Resolved that bug in master.

> Pentium 3558U 10.13 Doesn't work.

This looks like an issue with your 10.13 installation. The operating system does not check the signature integrity, so the patch does not apply. You may want to try reinstalling the operating system, ensuring that System Integrity Protection and Secure Boot are enabled, and possibly rebuilding dyld shared cache.

**Comment by whatnameisit** [comment]
I've installed High Sierra on an external disk and used that to test https://github.com/acidanthera/RestrictEvents/commit/a14224420015ec0b2c3d21d8a1fe04f858eaa9ce on Pentium 3558U first and then the i5 laptop. I enabled SIP on both systems, but not Apple Secure Boot when testing. I'll see if I get a different result this time with Secure Boot.
And thanks for fixing the spaces bug. I'll try it on Monday.
